## Sentiment Analysis using Valence Aware Dictionary for Sentiment Reasoner (VADER)

In [1]:
# Import VADER lexicon
import nltk
nltk.download('vader_lexicon')

[nltk_data] Downloading package vader_lexicon to
[nltk_data]     /Users/edgar/nltk_data...
[nltk_data]   Package vader_lexicon is already up-to-date!


True

In [2]:
# Import SentimentIntensityAnalyzer and create an sid object

from nltk.sentiment.vader import SentimentIntensityAnalyzer # use NLTK vader documentation

sid = SentimentIntensityAnalyzer() # sid object

In [3]:
# Create review as a string

review = 'I really loved the product. I would recommend this to family and friends.'

In [4]:
# Create review rating function
# Compound score provides polarity and overall sentiment

def review_rating(review):
    rating = sid.polarity_scores(review)    # Polarity Range: [-1,1], where 0 is Neutral. 
    if rating['compound'] < 0:              # Negative rating when rating<0
        print('Review Sentiment: Negative')
    elif rating['compound'] == 0:           # Neutral when rating = 0
        print('Review Sentiment: Neutral')
    else: 
        print('Review Sentiment: Positive') # positive when rating>0 (all other cases)
    print()
    print('Compound Score:', rating['compound'])

In [5]:
# Pass the review to the review_rating function

review_rating(review)

Review Sentiment: Positive

Compound Score: 0.8687


###### For the review used above, the sentiment was determined to be positive, with a compound score of 0.8687. The compound score is high on the positive polarity range. The review was written to contain a high positive sentiment and it coincides with VADER's sentiment classification of the review. 